Core/src/Vertexing/KalmanVertexUpdater.cpp

```diff
@@ -145,8 +160,14 @@ void update(Vector4& vtxPos, SquareMatrix4& vtxCov,
   KalmanVertexUpdater::Cache<nDimVertex> cache;
 
   // Calculate update and save result in cache
-  calculateUpdate(vtxPos, vtxCov, trk.linearizedState, trackWeight, sign,
-                  cache);
+
+  if constexpr (nDimVertex == 3) {
+    calculateUpdateImpl(vtxPos, vtxCov, trk.linearizedState, trackWeight, sign,
+                        cache);
+  } else if constexpr (nDimVertex == 4) {
+    calculateUpdateImpl(vtxPos, vtxCov, trk.linearizedState, trackWeight, sign,
+                        cache);
+  }
```

Comment:
why do we use `calculateUpdateImpl` instead of `calculateUpdate` now? is `calculateUpdate` obsolete now?

Reply:
It's called in two placed, but is called from another compilation unit.

I refactored this PR to merge the two CU, and with that, there only remain TWO `Impl` functions:

`updateVertexWithTrackImpl` and `updateTrackWithVertexImpl`

corresponding to the only two public functions that remain.